open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/common/idempotency/IdempotencyValidator.java

```diff
@@ -80,123 +79,146 @@ public static IdempotencyValidationResult validateIdempotency(String idempotency
-     * @param dbClientId          client ID retrieved from the database
-     * @return   Whether JSON client Ids are equal
+     * @param resourcePath     Resource Path
+     * @return idempotency Attribute Name.
      */
-    private static boolean isClientIdsMatching(String requestClientID, String dbClientId) {
-
-        return requestClientID.equals(dbClientId);
+    public String getIdempotencyAttributeName(String resourcePath) {
+        return "IdempotencyKey";
     }
 
     /**
-     * Method to compare whether JSON payloads are equal.
+     * Method to get the Idempotency Header Name according to the request.
      *
-     * @param jsonString1     JSON payload retrieved from database
-     * @param jsonString2     JSON payload received from current request
-     * @return   Whether JSON payloads are equal
-     * @throws IOException If an error occurs while comparing JSON payloads
+     * @return idempotency Header Name.
      */
-    private static boolean isJSONPayloadSimilar(String jsonString1, String jsonString2) throws IOException {
-
-        JsonNode expectedNode = new ObjectMapper().readTree(jsonString1);
-        JsonNode actualNode = new ObjectMapper().readTree(jsonString2);
-        return expectedNode.equals(actualNode);
+    public String getIdempotencyHeaderName() {
+        return "x-idempotency-key";
     }
 
     /**
-     * Method to check whether difference between two dates is less than the configured time.
+     * Method to get created time from the Detailed Consent Resource.
      *
-     * @param createdTime Created Time of the request
-     * @return  Whether the request is received within allowed time
+     * @param resourcePath     Resource Path
+     * @param consentId             ConsentId
+     * @return Created Time.
      */
-    protected static boolean isRequestReceivedWithinAllowedTime(long createdTime) {
-
-        if (createdTime == 0L) {
-            return false;
+    public long getCreatedTimeOfPreviousRequest(String resourcePath, String consentId) {
+        DetailedConsentResource consentRequest = null;
+        try {
+            consentRequest = consentCoreService.getDetailedConsent(consentId);
+        } catch (ConsentManagementException e) {
+            log.error(IdempotencyConstants.CONSENT_RETRIEVAL_ERROR, e);
+            return 0L;
         }
-        String allowedTimeDuration = (String) configs.get(IdempotencyConstants.IDEMPOTENCY_ALLOWED_TIME);
-        if (allowedTimeDuration != null) {
-            OffsetDateTime createdDate = OffsetDateTime.parse(convertToISO8601(createdTime));
-            OffsetDateTime currDate = OffsetDateTime.now(createdDate.getOffset());
 
-            long diffInHours = Duration.between(createdDate, currDate).toMinutes();
-            return diffInHours <= Long.parseLong(allowedTimeDuration);
-        } else {
-            log.error("Idempotency Allowed duration is null");
-            return false;
+        return consentRequest.getCreatedTime();
```

Comment:
Couldn't consentRequest be null?

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/51/commits/e1205691228fc4b76853a0eab7dd99dac2516661